### ISSUE TITLE
Using shadows now not only text-shadow but also drop-shadow causes preformance to drop significately

### ISSUE THREAD

**Issue by tomasts248** [GoogleChrome/lighthouse#15560]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://pagespeed.web.dev/analysis/https-www-ingenieroshs-com/4m9gz64b73?form_factor=mobile

### What happened?

As I learned in the past https://github.com/GoogleChrome/lighthouse/issues/15180
There are issues with GPU utilization and shadows  especially with multiple values but it used to be a solution to move from text-shadow and use drop-shadow (filter property), but now even that is causing a huge difference in performance in scores..
After removing CSS shadows with multiple values I'm getting:
https://pagespeed.web.dev/analysis/https-www-ingenieroshs-com/3ldfq73169?form_factor=desktop
while using them causes:
https://pagespeed.web.dev/analysis/https-www-ingenieroshs-com/4m9gz64b73?form_factor=mobile

Seems like even using single values affect performance.. but I need to further test this... But I'm worried about is there any expected solution in the middle / long term for this?

### What did you expect?

To perform better while using shadow CSS properties.

### What have you tried?

Removing CSS shadow rules and comparing PS scores.

### How were you running Lighthouse?

PageSpeed Insights

### Lighthouse Version

11.0.0

**Comment by adamraine** [comment]
Can you provide a version of the site with unexpectedly bad performance score so we can verify if this is the same GPU acceleration issue?

**Comment by tomasts248** [comment]
Sure.
Question: with text-shadow or drop-shadow filter property?

**Comment by adamraine** [comment]
Whichever reproduces the unexpected results in https://pagespeed.web.dev/analysis/https-www-ingenieroshs-com/4m9gz64b73?form_factor=mobile

I can't reproduce this issue on the current site.

**Comment by tomasts248** [comment]
Ok uploaded the drop-shadow rule with multiple values..
I'm getting the error at least form here:
https://pagespeed.web.dev/analysis/https-www-ingenieroshs-com/unjhbql65v?form_factor=desktop

**Comment by adamraine** [comment]
Taking a look, this seems like the same issue as https://github.com/GoogleChrome/lighthouse/issues/15180 and https://github.com/GoogleChrome/lighthouse/issues/8557

Here is a trace I collected running Lighthouse locally: https://trace.cafe/t/gxzOyBmns6

On PSI I'm getting `NO_FCP` so I suspect that the "Commit" rendering phase is even slower on the PSI backend such that the navigation times out waiting for FCP.

FWIW the responsiveness of the page is pretty bad on my end (at least with the version of the page published at time of writing this). I suspect the `drop-shadow` rule has a real impact on performance even if GPU acceleration is enabled.

**Comment by tomasts248** [comment]
> I suspect the drop-shadow rule has a real impact on performance even if GPU acceleration is enabled.

You mean it improves with GPU? or it is worse? I ask to be sure. 
If the second is true, why before PSI didn't account for drop-shadow replacing text-shadow in the scores? What caused the change?

**Comment by tomasts248** [comment]
Also going a bit off topic but do you happen to know where to report/suggest the idea for a fourth value similar to box-shadow (if I recap correctly is the spread value) but for text-shadow and drop-shadow (which have only 3 values aside from color) so that there's no need to create several values in one rule.. that could be  a proper solution?

**Comment by adamraine** [comment]
> You mean it improves with GPU? or it is worse? I ask to be sure.
If the second is true, why before PSI didn't account for drop-shadow after replacing text-shadow, in the scores? What caused the change?

The performance metrics (TBT in particular) improve when the GPU is used. However, the page still has responsiveness issues when analyzing the page qualitatively.

> Yep I used the most extreme of the impacting rules.. I think if I reduce the number of values at least will not throw an error, let me know if that is useful to further test or not really

I don't think you need to make further changes for testing. It is pretty clear to me at this point that this should be merged into https://github.com/GoogleChrome/lighthouse/issues/8557.

> Also going a bit off topic but do you happen to know where to report/suggest the idea for a fourth value similar to box-shadow (if I recap correctly is the spread value) but for text-shadow and drop-shadow (which have only 3 values aside from color) so that there's no need to create several values in one rule.. that could be a proper solution?

Not sure what you are referring to. I don't think this is the best place to answer specific questions about `text-shadow` and `drop-shadow`.

**Comment by tomasts248** [comment]
Ok, then I'm removin the problematic rules.

> However, the page still has responsiveness issues when analyzing the page qualitatively.

could you mind elaborating more on that? in order to fix them if you don't mind.

**Comment by paulirish** [comment]
> There are issues with GPU utilization and shadows especially with multiple values but it used to be a solution to move from text-shadow and use drop-shadow (filter property), but now even that is causing a huge difference in performance in scores..

two things are going on here.

## one: shadow cost

One. you're TOTALLY correct about gpu and multiple drop-shadows.  `filter` is one of the few css properties that gets handled on the GPU (along with `transform`, `opacity` and sometimes `clip`)

the GPU rendering costs for the `filter: drop-shadow` is indeed VERY high. surprisingly high.   

text-shadow and box-shadow are done on CPU instead of GPU.. while generally you'd think that shifting to GPU is good, in this case... `filter` is so unoptimized that i think you should avoid it entirely.   

For this specific visual effect, I'd go with multiple text-shadows. (yes a spread value would be useful for it, i agree.. \*googles a bit\* ....  it looks like the CSS WG has [granted your feature request](https://github.com/w3c/csswg-drafts/issues/6971), but its not implemented. (except in IE11 apparently!))

## two: gpu and pagespeed insights

the backend of PSI doesn't have GPUs so stuff like webgl is done in software instead. it gets over 10x slower. that is almost exactly whats happening to your `filter: drop-shadow()s` too.. So.. while we see those rules really slow down rendering in regular desktop.. on the PSI backend it gets so slow that it fails to finish rendering. (thus NO_FCP)

---------

tldr: don't use `filter` if you care about performance.  [some other folks exploring that..](https://www.google.com/search?q=filter+css+performance)

**Comment by paulirish** [comment]
> could you mind elaborating more on that? in order to fix them if you don't mind.

we don't normally handfeed performance insights  but.... since you were right about the shadow and GPU here's a few more things i see in the Performance panel:

just reloading the page it seems the TTFB on the html is like 2s. that's by far the biggest problem.

<img width="718" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/39191/dcb64110-2d1c-48ff-b7d3-ed3c063e0125">

all the render blocking css/js delay the first paint like 430ms:

<img width="755" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/39191/11f6c6b0-2c8e-47b8-9b75-5266cba0a22e">

also i see ~200ms TTFB on these static assets:

<img width="752" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/39191/68176314-1f30-44ff-a797-a2de17637a4e">


thus: you need caching on your webhost so it can serve the HTML and all the subresources immediately.  and you should try to get rid of the renderblocking requests. (move js to be async. inline any necessary css into HTML)

good luck

**Comment by tomasts248** [comment]
Thank you so much for confirming and explaining whats going on :+1: 

> we don't normally handfeed performance insights but.... since you were right about the shadow and GPU here's a few more things i see in the Performance panel:

Thanks for that!! appreciated, well we are  going to setup Cloudflare once the domain switch is done, so we will have HTML caching by Cloudflare too ;) which should overall improve performance a lot hope that solves those indicators if not I will have to further optimize it someway, but I we shall see.
